### PR TITLE
Fix demo alert for inbound email

### PR DESCRIPTION
**Pull request by vadimkerr**
# What this PR does
Fix HTTP 500 when sending demo alert for inbound integration email. 

- Make all integration configs have consistent `is_demo_alert_enabled` and `example_payload` values
- Add tests

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/alerts/models/alert_receive_channel.py [hunk 1]
```diff
@@ -519,42 +519,35 @@ def heartbeat_module(self):
     # Demo alerts
     def send_demo_alert(self, force_route_id=None, payload=None):
         logger.info(f"send_demo_alert integration={self.pk} force_route_id={force_route_id}")
+
+        if not self.is_demo_alert_enabled:
+            raise UnableToSendDemoAlert("Unable to send demo alert for this integration.")
+
         if payload is None:
             payload = self.config.example_payload
-        if self.is_demo_alert_enabled:
```
Comment (vadimkerr) [inline engine/apps/alerts/models/alert_receive_channel.py:524]: I tried to simplify things a bit and reduce number of indentation levels. 
The only logical change is checking for `if not self.is_demo_alert_enabled` before anything.

**Review by matiasb — APPROVED** [review]
LGTM

File: engine/apps/alerts/tests/test_default_templates.py [hunk 2]
```diff
@@ -102,3 +102,26 @@ def test_default_templates_are_valid():
                     jinja_template_env.from_string(template)
                 except TemplateSyntaxError as e:
                     pytest.fail(e.message)
+
+
+@pytest.mark.parametrize("config", AlertReceiveChannel._config)
```
Comment (joeyorlando) [inline engine/apps/alerts/tests/test_default_templates.py:107]: nice ✌️

File: engine/apps/alerts/tests/test_alert_receiver_channel.py [hunk 3]
```diff
@@ -145,6 +146,21 @@ def test_send_demo_alert_alertmanager_payload_shape(
     assert mocked_create_alert.call_args.args[1]["force_route_id"] is None
 
 
+@mock.patch("apps.integrations.tasks.create_alert.apply_async", return_value=None)
+@pytest.mark.parametrize(
+    "integration", [config.slug for config in AlertReceiveChannel._config if not config.is_demo_alert_enabled]
```
Comment (joeyorlando) [inline engine/apps/alerts/tests/test_alert_receiver_channel.py:151]: 👍

**Review by joeyorlando — APPROVED** [review]
LGTM! You might want to double check this integration file [here](https://github.com/grafana/oncall-private/blob/dev/engine_enterprise/config_integrations/demo.py) in the private repo

**Review by iskhakov — APPROVED** [review]

**Pull request merged**